IRremoteESP8266.cpp

```diff
@@ -702,23 +702,32 @@ static void ICACHE_RAM_ATTR read_timeout(void *arg __attribute__((unused))) {
 
 static void ICACHE_RAM_ATTR gpio_intr() {
   uint32_t gpio_status = GPIO_REG_READ(GPIO_STATUS_ADDRESS);
+  uint32_t now = system_get_time();
+  static uint32_t start = 0;
+
+  os_timer_disarm(&timer);
   GPIO_REG_WRITE(GPIO_STATUS_W1TC_ADDRESS, gpio_status);
 
-	if (irparams.rcvstate == STATE_STOP) {
-		return;
-	}
-  static uint32_t start = 0;
-  uint32_t now = system_get_time();
-	if (irparams.rcvstate == STATE_IDLE) {
-		irparams.rcvstate = STATE_MARK;
-		irparams.rawbuf[irparams.rawlen++] = 20;
-	} else if (irparams.rawlen < RAWBUF) {
-		irparams.rawbuf[irparams.rawlen++] = (now - start) / USECPERTICK + 1;
+  if (irparams.rawlen >= RAWBUF) {
+    irparams.overflow = true;
+    irparams.rcvstate = STATE_STOP;
   }
-  start = now;
 
-  os_timer_disarm(&timer);
-  os_timer_arm(&timer, 15, 0);
+  if (irparams.rcvstate == STATE_STOP) {
+    return;
+  }
+
+  if (irparams.rcvstate == STATE_IDLE) {
+    irparams.overflow = false;
+    irparams.rcvstate = STATE_MARK;
+    irparams.rawbuf[irparams.rawlen++] = 1;
+  } else {
+    irparams.rawbuf[irparams.rawlen++] = (now - start) / USECPERTICK + 1;
```

Comment:
There is a small near but non-zero chance that "now" can be less that "start". i.e. when the microsecond counter wraps.
This happens after 71.56mins.
We can accept that as a known risk, if so, lets add a comment to that effect, or we can detect the wrap-around of the counter.
something like (not tested):
if (now < start)
  irparams.rawbuf[irparams.rawlen++] = (0xFFFFFFFF - start + now) / USECPERTICK + 1;
else
  <existing line>
...